File: Projects/ZZZ_RecylerViewTwo/app/src/main/java/com/nsw2022/zzz_recylerviewtwo/MyAdapter.java
package com.nsw2022.zzz_recylerviewtwo;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.VH>{
  Context context;
  ArrayList<Item> items;

    public MyAdapter(Context context, ArrayList<Item> items) {
        this.context = context;
        this.items = items;
    }

    public MyAdapter() {
    }

    @NonNull
    @Override
    public VH onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater=LayoutInflater.from(context);
        View itemView=layoutInflater.inflate(R.layout.recyclerview_item,parent,false);

        VH holder = new VH(itemView);

        return holder;
    }

    @Override
    public void onBindViewHolder(@NonNull VH holder, int position) {

        Item item=items.get(position);

        holder.tv_title.setText(item.tv_title);
        holder.tv_price.setText(item.tv_price);

        Glide.with(context).load(item.iv).into(holder.imgID);
        Glide.with(context).load(item.hertid).into(holder.imgheart);
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    class VH extends RecyclerView.ViewHolder{

        ImageView imgID,imgheart;
        TextView tv_title, tv_price;

        public VH(@NonNull View itemView) {
            super(itemView);
            imgID= itemView.findViewById(R.id.iv);
            imgheart=itemView.findViewById(R.id.herat);

            tv_title=itemView.findViewById(R.id.tv_title);
            tv_price=itemView.findViewById(R.id.price);

        }
    }


}//MyAdapter

File: Projects/Ex91KotlinRecyclerView/app/src/main/java/com/nsw2022/ex91kotlinrecyclerview/MyAdapter.kt
package com.nsw2022.ex91kotlinrecyclerview

import android.content.Context
import android.icu.lang.UCharacter.GraphemeClusterBreak.V
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide

class MyAdapter constructor(val context: Context,var items: MutableList<Item>): RecyclerView.Adapter<MyAdapter.VH>(){/////////////////////


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): VH {
        var layoutInflater: LayoutInflater = LayoutInflater.from(context)
        val itemView:View = layoutInflater.inflate(R.layout.recycler_item,parent,false)
        return VH(itemView)
    }

    override fun onBindViewHolder(holder: VH, position: Int) {
        val item:Item = items.get(position)

        holder.tvTitle.text = item.title
        //코틀린 언어는 setXXX()를 싫어함. 그냥 XXX를 변숴럼 - 대입연산자로 값 설정을 권장
        holder.tvMSG.text = item.msg
        Glide.with(context).load(item.img).into(holder.iv)

    }

    //함수의 return 코드 단순화 [ 할당연산자= ]
    override fun getItemCount(): Int= items.size


    inner class VH constructor(itemView: View) : RecyclerView.ViewHolder(itemView){
        // itemView 의 자식뷰들을 참조
        val tvTitle:TextView by lazy { itemView.findViewById(R.id.tv_title) }
        val tvMSG:TextView by lazy { itemView.findViewById(R.id.tv_msg) }
        val iv:ImageView by lazy { itemView.findViewById(R.id.iv) }
    }
}